include/seqan/stream/iostream_zip.h

```diff
@@ -232,408 +188,278 @@ class basic_unzip_streambuf :
-        // taken care of during the last zflush_finalize()
-        // called by the higher level close() routines
-        zflush_finalize();
-    }
+        this->flush(); this->rdbuf()->flush_finalize();
 
-    /// @brief returns true if it is a gzip
-    bool is_gzip() const { return m_is_gzip; }
-
-    /// @brief flush inner buffer and zipper buffer
-    basic_zip_ostream< Elem, Tr > &
-    zflush()
-    {
-        flush(); rdbuf()->flush(); return *this;
-    }
-
-    /// @brief flush inner and zipper buffers and finalize zip stream
-    basic_zip_ostream< Elem, Tr > &
-    zflush_finalize()
-    {
-
-        flush(); rdbuf()->flush_finalize();
-
-        if ( m_is_gzip && ( rdbuf()->get_zerr() == Z_STREAM_END ) && ( !m_zip_stream_finalized ) ) {
-            add_footer(); // writes crc trailer to end the current zip stream
-            flush();
-            m_zip_stream_finalized = true;
+        if (m_is_gzip && (this->rdbuf()->get_zerr() == Z_STREAM_END))
+        {
+            add_footer();
+            this->flush();
         }
-
-        return *this;
-    }
-
-    /// @brief stream output
-    /// @details if zip stream has been finalized, will reset
-    /// @details the stream and add header if necessary
-    template< typename T >
-    inline
-    basic_zip_ostream &
-    operator <<( T const & t )
-    {
-        reset_zip_stream();
-        static_cast< std::ostream & >( *this ) << t;
-        return *this;
     }
 
-    /// @brief write char
-    /// @details if zip stream has been finalized, will reset
-    /// @details the stream and add header if necessary
-    inline
-    basic_zip_ostream &
-    put( char const c )
-    {
-        reset_zip_stream();
-        static_cast< std::ostream & >( *this ).put( c );
-        return *this;
-    }
+    // returns true if it is a gzip
+    bool is_gzip() const        { return m_is_gzip; }
 
-    /// @brief write a string
-    /// @details if zip stream has been finalized, will reset
-    /// @details the stream and add header if necessary
-    inline
-    basic_zip_ostream &
-    write( char const * str, std::streamsize const count )
+    // flush inner buffer and zipper buffer
+    basic_zip_ostream<Elem, Tr> & zflush()
     {
-        reset_zip_stream();
-        static_cast< std::ostream & >( *this ).write( str, count );
-        return *this;
+        this->flush(); this->rdbuf()->flush(); return *this;
     }
 
 private:
-
-    /// @brief if end of stream, reset the zip stream and add header
-    inline
-    bool
-    reset_zip_stream()
-    {
-        if ( rdbuf()->get_zerr() == Z_STREAM_END ) {
-            rdbuf()->reset_state();
-            add_header();
-            m_zip_stream_finalized = false;
-            return true;
-        }
-        return false;
-    }
-
-    static void put_long_as_uint32( ostream_reference out_, unsigned long x_ );
-
+    void put_uint32(unsigned long x_);
     void add_header();
```

Comment:
either use fixed size integer types or not, but don't mix. unsigned long is 64bit on 64bit linux and mac!